src/executor/admin/SubmitJobExecutor.cpp

```diff
@@ -18,20 +18,35 @@ folly::Future<Status> SubmitJobExecutor::execute() {
 
     auto *sjNode = asNode<SubmitJob>(node());
     auto jobOp = sjNode->jobOp();
-    meta::cpp2::AdminCmd cmd = meta::cpp2::AdminCmd::COMPACT;
+    std::vector<std::string> jobArguments;
+    auto spaceName = qctx()->rctx()->session()->space().name;
+    jobArguments.emplace_back(std::move(spaceName));
+
+    meta::cpp2::AdminCmd cmd;
     if (jobOp == meta::cpp2::AdminJobOp::ADD) {
-        std::vector<std::string> params;
-        folly::split(" ", sjNode->params().front(), params, true);
-        if (params.front() == "compact") {
+        auto& params = sjNode->params();
+        if (params[0] == "compact") {
             cmd = meta::cpp2::AdminCmd::COMPACT;
-        } else if (params.front() == "flush") {
+        } else if (params[0] == "flush") {
             cmd = meta::cpp2::AdminCmd::FLUSH;
+        } else if (params[0] == "rebuild") {
+            if (params[1] == "tag") {
+                cmd = meta::cpp2::AdminCmd::REBUILD_TAG_INDEX;
+                jobArguments.emplace_back(params[3]);
```

Comment:
I think the handle needs to move to validator because needs to check the index if exists.